oshi-core/src/main/java/oshi/software/common/AbstractOperatingSystem.java

```diff
@@ -170,29 +163,36 @@ protected static void addChildrenToDescendantSet(Collection<OSProcess> allProcs,
      *            as value
      * @param parentPid
      *            The process ID whose children or descendants to return
-     * @param descendantPids
-     *            On input, an empty set, or a set of other children already
-     *            retrieved when using recursion. On output, the children of the
-     *            parent added to the set.
-     * @param recurse
+     * @param allDescendants
      *            If false, only gets immediate children of this process. If true,
      *            gets all descendants.
+     * @return Set of children or descendants of parentPid
      */
-    protected static void addChildrenToDescendantSet(Map<Integer, Integer> parentPidMap, int parentPid,
-            Set<Integer> descendantPids, boolean recurse) {
+    protected static Set<Integer> getChildrenOrDescendants(Map<Integer, Integer> parentPidMap, int parentPid,
+            boolean allDescendants) {
+        Set<Integer> descendantPids = new HashSet();
+        Queue<Integer> queue = new LinkedList();
```

Comment:
Not a big deal, but I think `ArrayDeque` may be a faster and more memory efficient `Queue` implementation.  

Rule of thumb: nearly always*, array implementations have better performance than linked lists.
* Unless you have a need to remove elements while iterating, which is not the case here.